pkg/pillar/devicenetwork/dnc.go

```diff
@@ -246,7 +246,12 @@ func VerifyPending(ctx *DeviceNetworkContext, pending *DPCPending,
 	log.Infof("VerifyPending: No required ports missing. " +
 		"parsing device port config list")
 
-	if !pending.PendDPC.MostlyEqual(&pending.OldDPC) {
+	// When there are port level errors in DPC, we should re-test the
+	// existing DPC (even when the old and new DPC are same). There could
+	// have been a delay in the port being available in kernel or other
+	// transient errors during last test. We should re-test the DPC to see
+	// if the previous errors are gone and the setup improves.
+	if !pending.PendDPC.MostlyEqual(&pending.OldDPC) || ctx.DeviceNetworkStatus.HasErrors() {
```

Comment:
This block is very subtle around DHCP I've we've broken and fixed it several times.

The key is OldDPC. It must match the current running dhcpcd configuration, since we rely on that comparison to determine which ones need to be activated, inactivated, or modified (inactivate then activate with a different config). 

I don't see how this and your other DHCP changes ensure that OldDPC always tracks which dhcpcd are currently runnig.

What bug requires these dhcpcd-related fixes?

Reply:
We currently do not handle the case where an interfaces gets deleted and added back. Today we only start dhcpcd as part of the DPC verification process. As part of the current change we do not select a new DPC. We just try to re-start verification on the same DPC at index 0.